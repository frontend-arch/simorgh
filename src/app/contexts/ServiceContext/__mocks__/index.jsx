import React from 'react';
import { node, string } from 'prop-types';
import services from '#server/utilities/serviceConfigs';

/*
 * This file is mocked by default to avoid having to handle
 * async behavior in tests across the application when
 * using service contexts.
 */
export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => (
  <ServiceContext.Provider value={services[service][variant || 'default']}>
    {children}
  </ServiceContext.Provider>
);

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string.isRequired,
  variant: string,
};

ServiceContextProvider.defaultProps = {
  variant: 'default',
};
