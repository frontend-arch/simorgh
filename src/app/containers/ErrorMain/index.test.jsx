import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import ErrorMain from './index';

describe('ErrorMain', () => {
  shouldMatchSnapshot(
    'should correctly render for 404',
    <ServiceContextProvider service="news">
      <ErrorMain status={404} />
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should correctly render for 500',
    <ServiceContextProvider service="news">
      <ErrorMain status={500} />
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should correctly render for other status code',
    <ServiceContextProvider service="news">
      <ErrorMain status={123} />
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should correctly render for 404 for persian',
    <ServiceContextProvider service="persian">
      <ErrorMain status={404} />
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should correctly render for 500 for persian',
    <ServiceContextProvider service="persian">
      <ErrorMain status={500} />
    </ServiceContextProvider>,
  );
});
