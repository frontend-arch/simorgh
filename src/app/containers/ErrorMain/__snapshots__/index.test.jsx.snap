// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMain should correctly render for 404 1`] = `
.c5 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.c5:focus,
.c5:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

.c0 {
  margin: 0 auto;
  background: #FDFDFD;
  padding-bottom: 2rem;
}

.c2 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #B80000;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
}

.c3 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #3F3F42;
  margin-top: 0;
}

.c1 {
  grid-column: 1 / span 6;
  padding-bottom: 4rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
  padding-top: 0.2rem;
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.375rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

<html
  dir="ltr"
  lang="en-GB"
>
  <head>
    <title>
      Page cannot be found - BBC News
    </title>
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noindex,nofollow"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="Page cannot be found"
      name="og:description"
    />
    <meta
      content="Page cannot be found - BBC News"
      name="og:title"
    />
    <meta
      content="Page cannot be found"
      name="twitter:description"
    />
    <meta
      content="Page cannot be found - BBC News"
      name="twitter:title"
    />
  </head>
  <body>
    <div>
      <main
        role="main"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <span
              class="c2"
            >
              404
            </span>
            <h1
              class="c3"
              id="content"
              tabindex="-1"
            >
              Page cannot be found
            </h1>
            <p
              class="c4"
            >
              Sorry, we're unable to bring you the page you're looking for. Please try:
            </p>
            <ul>
              <li
                class="c4"
              >
                Double checking the url
              </li>
              <li
                class="c4"
              >
                Hitting the refresh button in your browser
              </li>
              <li
                class="c4"
              >
                Searching for this page using the BBC search bar
              </li>
            </ul>
            <p
              class="c4"
            >
              Alternatively, please visit the 
              <a
                class="c5"
                href="https://www.bbc.com/news"
              >
                BBC News homepage.
              </a>
              
            </p>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;

exports[`ErrorMain should correctly render for 404 for persian 1`] = `
<html
  dir="rtl"
  lang="fa"
>
  <head>
    <title>
      این صفحه پیدا نشد - BBC News فارسی
    </title>
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noindex,nofollow"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="این صفحه پیدا نشد"
      name="og:description"
    />
    <meta
      content="این صفحه پیدا نشد - BBC News فارسی"
      name="og:title"
    />
    <meta
      content="این صفحه پیدا نشد"
      name="twitter:description"
    />
    <meta
      content="این صفحه پیدا نشد - BBC News فارسی"
      name="twitter:title"
    />
  </head>
  <body>
    .c5 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.c5:focus,
.c5:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

.c0 {
  margin: 0 auto;
  background: #FDFDFD;
  padding-bottom: 2rem;
}

.c2 {
  font-size: 1.5rem;
  line-height: 1.875rem;
  color: #B80000;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
}

.c3 {
  font-size: 2rem;
  line-height: 2.625rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #3F3F42;
  margin-top: 0;
}

.c1 {
  grid-column: 1 / span 6;
  padding-bottom: 4rem;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
  padding-top: 0.2rem;
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 2rem;
    line-height: 2.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2.375rem;
    line-height: 2.875rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3.375rem;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.375rem;
    line-height: 2rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.375rem;
    line-height: 2rem;
  }
}

<div>
      <main
        role="main"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <span
              class="c2"
            >
              ۴۰۴
            </span>
            <h1
              class="c3"
              id="content"
              tabindex="-1"
            >
              این صفحه پیدا نشد
            </h1>
            <p
              class="c4"
            >
              متاسفانه صفحه مورد نظر شما را پیدا نمی‌کنیم. لطفا گزینه‌های زیر را امتحان کنید:
            </p>
            <ul>
              <li
                class="c4"
              >
                آدرس صفحه مورد نظر را دوباره بررسی کنید
              </li>
              <li
                class="c4"
              >
                با مرورگر خود صفحه را ریفرش (Refresh) کنید
              </li>
              <li
                class="c4"
              >
                در نوار جستجوی بی‌بی‌سی دنبال اطلاعات مورد نظرتان بگردید
              </li>
            </ul>
            <p
              class="c4"
            >
              برای یافتن اطلاعات مورد نظر به 
              <a
                class="c5"
                href="https://www.bbc.com/persian"
              >
                صفحه اصلی بی‌بی‌سی فارسی
              </a>
               بروید
            </p>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;

exports[`ErrorMain should correctly render for 500 1`] = `
<html
  dir="ltr"
  lang="en-GB"
>
  <head>
    <title>
      Internal server error - BBC News
    </title>
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noindex,nofollow"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="Internal server error"
      name="og:description"
    />
    <meta
      content="Internal server error - BBC News"
      name="og:title"
    />
    <meta
      content="Internal server error"
      name="twitter:description"
    />
    <meta
      content="Internal server error - BBC News"
      name="twitter:title"
    />
  </head>
  <body>
    .c5 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.c5:focus,
.c5:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

.c0 {
  margin: 0 auto;
  background: #FDFDFD;
  padding-bottom: 2rem;
}

.c2 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #B80000;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
}

.c3 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #3F3F42;
  margin-top: 0;
}

.c1 {
  grid-column: 1 / span 6;
  padding-bottom: 4rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
  padding-top: 0.2rem;
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.375rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

<div>
      <main
        role="main"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <span
              class="c2"
            >
              500
            </span>
            <h1
              class="c3"
              id="content"
              tabindex="-1"
            >
              Internal server error
            </h1>
            <p
              class="c4"
            >
              Sorry, we're currently unable to bring you the page you're looking for. Please try:
            </p>
            <ul>
              <li
                class="c4"
              >
                Hitting the refresh button in your browser
              </li>
              <li
                class="c4"
              >
                Coming back again later
              </li>
            </ul>
            <p
              class="c4"
            >
              Alternatively, please visit the 
              <a
                class="c5"
                href="https://www.bbc.com/news"
              >
                BBC News homepage.
              </a>
              
            </p>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;

exports[`ErrorMain should correctly render for 500 for persian 1`] = `
<html
  dir="rtl"
  lang="fa"
>
  <head>
    <title>
      خطا در سرور داخلی - BBC News فارسی
    </title>
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noindex,nofollow"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="خطا در سرور داخلی"
      name="og:description"
    />
    <meta
      content="خطا در سرور داخلی - BBC News فارسی"
      name="og:title"
    />
    <meta
      content="خطا در سرور داخلی"
      name="twitter:description"
    />
    <meta
      content="خطا در سرور داخلی - BBC News فارسی"
      name="twitter:title"
    />
  </head>
  <body>
    .c5 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.c5:focus,
.c5:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

.c0 {
  margin: 0 auto;
  background: #FDFDFD;
  padding-bottom: 2rem;
}

.c2 {
  font-size: 1.5rem;
  line-height: 1.875rem;
  color: #B80000;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
}

.c3 {
  font-size: 2rem;
  line-height: 2.625rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #3F3F42;
  margin-top: 0;
}

.c1 {
  grid-column: 1 / span 6;
  padding-bottom: 4rem;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
  padding-top: 0.2rem;
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 2rem;
    line-height: 2.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2.375rem;
    line-height: 2.875rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3.375rem;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.375rem;
    line-height: 2rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.375rem;
    line-height: 2rem;
  }
}

<div>
      <main
        role="main"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <span
              class="c2"
            >
              ۵۰۰
            </span>
            <h1
              class="c3"
              id="content"
              tabindex="-1"
            >
              خطا در سرور داخلی
            </h1>
            <p
              class="c4"
            >
              متاسفانه صفحه مورد نظر شما در دسترس نیست. لطفا گزینه‌های زیر را امتحان کنید:
            </p>
            <ul>
              <li
                class="c4"
              >
                با مرورگر خود صفحه را ریفرش (Refresh) کنید
              </li>
              <li
                class="c4"
              >
                بعدا دوباره امتحان کنید
              </li>
            </ul>
            <p
              class="c4"
            >
              برای یافتن اطلاعات مورد نظر به 
              <a
                class="c5"
                href="https://www.bbc.com/persian"
              >
                صفحه اصلی بی‌بی‌سی فارسی
              </a>
               بروید
            </p>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;

exports[`ErrorMain should correctly render for other status code 1`] = `
<html
  dir="ltr"
  lang="en-GB"
>
  <head>
    <title>
      Internal server error - BBC News
    </title>
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noindex,nofollow"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="Internal server error"
      name="og:description"
    />
    <meta
      content="Internal server error - BBC News"
      name="og:title"
    />
    <meta
      content="Internal server error"
      name="twitter:description"
    />
    <meta
      content="Internal server error - BBC News"
      name="twitter:title"
    />
  </head>
  <body>
    .c5 {
  color: #222222;
  border-bottom: 1px solid #B80000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:visited {
  color: #6E6E73;
  border-bottom: 1px solid #6E6E73;
}

.c5:focus,
.c5:hover {
  border-bottom: 2px solid #B80000;
  color: #B80000;
}

.c0 {
  margin: 0 auto;
  background: #FDFDFD;
  padding-bottom: 2rem;
}

.c2 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #B80000;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
}

.c3 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #3F3F42;
  margin-top: 0;
}

.c1 {
  grid-column: 1 / span 6;
  padding-bottom: 4rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
  padding-top: 0.2rem;
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.375rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

<div>
      <main
        role="main"
      >
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <span
              class="c2"
            >
              500
            </span>
            <h1
              class="c3"
              id="content"
              tabindex="-1"
            >
              Internal server error
            </h1>
            <p
              class="c4"
            >
              Sorry, we're currently unable to bring you the page you're looking for. Please try:
            </p>
            <ul>
              <li
                class="c4"
              >
                Hitting the refresh button in your browser
              </li>
              <li
                class="c4"
              >
                Coming back again later
              </li>
            </ul>
            <p
              class="c4"
            >
              Alternatively, please visit the 
              <a
                class="c5"
                href="https://www.bbc.com/news"
              >
                BBC News homepage.
              </a>
              
            </p>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;
